torch_ipex/csrc/cpu/dbl/Common.cpp

```diff
@@ -17,9 +17,14 @@ namespace dbl {
 namespace comm {
 
 dil::tensor dil_tensor_from_dense(const at::Tensor& tensor) {
-  AT_ASSERTM(
+  TORCH_CHECK(
     tensor.layout() == at::Layout::Strided,
     "dil_tensor_view_from_dense expects dense tensor input");
+  TORCH_CHECK(tensor.scalar_type() == at::ScalarType::Float ||
+             tensor.scalar_type() == at::ScalarType::BFloat16,
+             "dil_tensor_view_from_dense expects bfloat16 or float tensor input");
+  TORCH_CHECK(tensor.dim() <= 5,
+             "Can't convert cpu tensor with the number of dimensions > 5");
```

Comment:
Could you please try to remove this assertion in transpose and verify if the `test_torch` passes?

https://github.com/intel/intel-extension-for-pytorch/blob/e91c403d19699f5832ce12c460634bfdbc556aec/torch_ipex/csrc/cpu/DevOPs.cpp#L1093

I think this PR should cover that case.